ClassCastException: class clojure.lang.Keyword cannot be cast to class java.lang.Throwable

### Describe the bug

During startup a ClassCastException is thrown. This is the stacktrace.

```
2024-02-07 15:16:10,482 DEBUG toucan2.query :: Apply kv-args {:is_sample true}
2024-02-07 15:16:10,514 ERROR metabase.core :: Metabase Initialization FAILED
clojure.lang.ExceptionInfo: class clojure.lang.Keyword cannot be cast to class java.lang.Throwable (clojure.lang.Keyword is in unnamed module of loader 'app'; java.lang.Throwable is in module java.base of loader 'bootstrap') {:toucan2/context-trace [["with resolved query" {:toucan2.pipeline/resolved-query {}}] ["with parsed args" {:toucan2.pipeline/query-type :toucan.query-type/select.instances, :toucan2.pipeline/parsed-args {:kv-args {:is_sample true}, :queryable {}}}] ["with model" {:toucan2.pipeline/model :model/Database}] ["with unparsed args" {:toucan2.pipeline/query-type :toucan.query-type/select.instances, :toucan2.pipeline/unparsed-args (:model/Database :is_sample true)}]]}
	at clojure.tools.logging$eval136$fn__139.invoke(NO_SOURCE_FILE:0)
	at clojure.tools.logging.impl$fn__1684$G__1666__1695.invoke(impl.clj:16)
	at clojure.tools.logging$log_STAR_.invokeStatic(logging.clj:63)
	at clojure.tools.logging$log_STAR_.invoke(logging.clj:42)
	at toucan2.honeysql2$apply_kv_arg_primary_method_default_clojure_lang_IPersistentMap_default.invokeStatic(honeysql2.clj:47)
	at toucan2.honeysql2$apply_kv_arg_primary_method_default_clojure_lang_IPersistentMap_default.invoke(honeysql2.clj:44)
	at clojure.lang.AFn.applyToHelper(AFn.java:171)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:675)
	at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
	at clojure.lang.RestFn.applyTo(RestFn.java:146)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at toucan2.tools.transformed$apply_kv_arg_primary_method_toucan2_tools_transformed_transformed_model_default_default.invokeStatic(transformed.clj:169)
	at toucan2.tools.transformed$apply_kv_arg_primary_method_toucan2_tools_transformed_transformed_model_default_default.invoke(transformed.clj:158)
	at clojure.lang.AFn.applyToHelper(AFn.java:171)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:675)
	at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
	at clojure.lang.RestFn.applyTo(RestFn.java:146)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at methodical.impl.combo.threaded$fn__17899$fn__17900$fn__17905.invoke(threaded.clj:72)
	at methodical.impl.combo.threaded$reducer_fn$fn__17869$fn__17873.invoke(threaded.clj:23)
	at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
	at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
	at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
	at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
	at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6886)
	at clojure.core$reduce.invoke(core.clj:6868)
	at methodical.impl.combo.threaded$reducer_fn$fn__17869.invoke(threaded.clj:21)
	at clojure.core$comp$fn__5876.doInvoke(core.clj:2589)
	at clojure.lang.RestFn.invoke(RestFn.java:467)
	at methodical.impl.combo.threaded$combine_with_threader$fn__17879.invoke(threaded.clj:45)
	at clojure.lang.AFn.applyToHelper(AFn.java:165)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:61)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:197)
	at toucan2.query$apply_kv_args$fn__21058.invoke(query.clj:222)
	at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49)
	at clojure.core.protocols$fn__8230.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8230.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6886)
	at clojure.core$reduce.invoke(core.clj:6868)
	at toucan2.query$apply_kv_args.invokeStatic(query.clj:220)
	at toucan2.query$apply_kv_args.invoke(query.clj:216)
	at toucan2.honeysql2$build_primary_method_default_default_clojure_lang_IPersistentMap.invokeStatic(honeysql2.clj:117)
	at toucan2.honeysql2$build_primary_method_default_default_clojure_lang_IPersistentMap.invoke(honeysql2.clj:111)
	at clojure.lang.AFn.applyToHelper(AFn.java:171)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:675)
	at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
	at clojure.lang.RestFn.applyTo(RestFn.java:146)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at toucan2.honeysql2$build_primary_method_toucan_query_type_select__STAR__default_clojure_lang_IPersistentMap.invokeStatic(honeysql2.clj:138)
	at toucan2.honeysql2$build_primary_method_toucan_query_type_select__STAR__default_clojure_lang_IPersistentMap.invoke(honeysql2.clj:120)
	at clojure.lang.AFn.applyToHelper(AFn.java:171)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:675)
	at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
	at clojure.lang.RestFn.applyTo(RestFn.java:146)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at methodical.impl.combo.threaded$fn__17899$fn__17900$fn__17905.invoke(threaded.clj:72)
	at methodical.impl.combo.threaded$reducer_fn$fn__17869$fn__17873.invoke(threaded.clj:23)
	at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
	at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
	at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
	at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
	at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6886)
	at clojure.core$reduce.invoke(core.clj:6868)
	at methodical.impl.combo.threaded$reducer_fn$fn__17869.invoke(threaded.clj:21)
	at clojure.core$comp$fn__5876.doInvoke(core.clj:2589)
	at clojure.lang.RestFn.invoke(RestFn.java:467)
	at methodical.impl.combo.threaded$combine_with_threader$fn__17879.invoke(threaded.clj:45)
	at clojure.lang.AFn.applyToHelper(AFn.java:165)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at metabase.db.setup$build_around_method_default.invokeStatic(setup.clj:205)
	at metabase.db.setup$build_around_method_default.invoke(setup.clj:198)
	at clojure.lang.AFn.applyToHelper(AFn.java:171)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:675)
	at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
	at clojure.lang.RestFn.applyTo(RestFn.java:146)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:61)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:197)
	at clojure.lang.Var.invoke(Var.java:399)
	at toucan2.pipeline$transduce_query_primary_method_default.invokeStatic(pipeline.clj:271)
	at toucan2.pipeline$transduce_query_primary_method_default.invoke(pipeline.clj:269)
	at clojure.lang.AFn.applyToHelper(AFn.java:178)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:675)
	at clojure.core$partial$fn__5908.doInvoke(core.clj:2639)
	at clojure.lang.RestFn.applyTo(RestFn.java:146)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at methodical.impl.combo.threaded$fn__17899$fn__17900$fn__17907.invoke(threaded.clj:79)
	at methodical.impl.combo.threaded$reducer_fn$fn__17869$fn__17873.invoke(threaded.clj:23)
	at clojure.lang.ArrayChunk.reduce(ArrayChunk.java:58)
	at clojure.core.protocols$fn__8244.invokeStatic(protocols.clj:136)
	at clojure.core.protocols$fn__8244.invoke(protocols.clj:124)
	at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
	at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6886)
	at clojure.core$reduce.invoke(core.clj:6868)
	at methodical.impl.combo.threaded$reducer_fn$fn__17869.invoke(threaded.clj:21)
	at clojure.core$comp$fn__5876.doInvoke(core.clj:2589)
	at clojure.lang.RestFn.applyTo(RestFn.java:146)
	at clojure.core$apply.invokeStatic(core.clj:675)
	at clojure.core$apply.doInvoke(core.clj:662)
	at clojure.lang.RestFn.invoke(RestFn.java:533)
	at methodical.impl.combo.threaded$combine_with_threader$fn__17879.doInvoke(threaded.clj:46)
	at clojure.lang.RestFn.applyTo(RestFn.java:151)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:675)
	at clojure.core$apply.doInvoke(core.clj:662)
	at clojure.lang.RestFn.invoke(RestFn.java:533)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
	at methodical.impl.standard$invoke_multifn.doInvoke(standard.clj:47)
	at clojure.lang.RestFn.invoke(RestFn.java:594)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:199)
	at toucan2.pipeline$transduce_query_STAR_.invokeStatic(pipeline.clj:278)
	at toucan2.pipeline$transduce_query_STAR_.invoke(pipeline.clj:274)
	at toucan2.pipeline$transduce_with_model.invokeStatic(pipeline.clj:293)
	at toucan2.pipeline$transduce_with_model.invoke(pipeline.clj:280)
	at toucan2.pipeline$transduce_parsed.invokeStatic(pipeline.clj:309)
	at toucan2.pipeline$transduce_parsed.invoke(pipeline.clj:295)
	at toucan2.pipeline$transduce_unparsed.invokeStatic(pipeline.clj:317)
	at toucan2.pipeline$transduce_unparsed.invoke(pipeline.clj:311)
	at toucan2.select$select_one.invokeStatic(select.clj:58)
	at toucan2.select$select_one.doInvoke(select.clj:50)
	at clojure.lang.RestFn.invoke(RestFn.java:436)
	at metabase.sample_data$update_sample_database_if_needed_BANG_.invokeStatic(sample_data.clj:88)
	at metabase.sample_data$update_sample_database_if_needed_BANG_.invoke(sample_data.clj:85)
	at metabase.core$init_BANG__STAR_.invokeStatic(core.clj:138)
	at metabase.core$init_BANG__STAR_.invoke(core.clj:98)
	at metabase.core$init_BANG_.invokeStatic(core.clj:156)
	at metabase.core$init_BANG_.invoke(core.clj:151)
	at metabase.core$start_normally.invokeStatic(core.clj:168)
	at metabase.core$start_normally.invoke(core.clj:162)
	at metabase.core$entrypoint.invokeStatic(core.clj:201)
	at metabase.core$entrypoint.doInvoke(core.clj:195)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
	at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.bootstrap.main(Unknown Source)
Caused by: java.lang.ClassCastException: class clojure.lang.Keyword cannot be cast to class java.lang.Throwable (clojure.lang.Keyword is in unnamed module of loader 'app'; java.lang.Throwable is in module java.base of loader 'bootstrap')
	... 190 more
2024-02-07 15:16:10,545 INFO metabase.core :: Metabase Shutting Down ...
```

### To Reproduce

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error


### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
Metabase v0.48.5
H2 database
```


### Severity

blocking

### Additional context

_No response_

I'm able to reproduce on 0.48.3 in addition to 0.48.5:

debug.xml
```
<?xml version="1.0" encoding="UTF-8"?>
<Configuration>
  <Appenders>
    <Console name="STDOUT" target="SYSTEM_OUT" follow="true">
      <PatternLayout pattern="%date %level %logger{2} :: %message%n%throwable">
        <replace regex=":basic-auth \\[.*\\]" replacement=":basic-auth [redacted]"/>
      </PatternLayout>
    </Console>

  </Appenders>
  <Loggers>
    <Root level="DEBUG">
      <AppenderRef ref="STDOUT"/>
    </Root>
  </Loggers>
</Configuration>
```

```shell
MB_JETTY_PORT=3006 java -Dlog4j2.configurationFile=debug.xml -jar 0.48.5.jar
...
2024-02-07 11:00:17,195 ERROR metabase.core :: Metabase Initialization FAILED
...
Caused by: clojure.lang.ExceptionInfo: class clojure.lang.Keyword cannot be cast to class java.lang.Throwable (clojure.lang.Keyword is in unnamed module of loader 'app'; java.lang.Throwable is in module java.base of loader 'bootstrap') {:toucan2/context-trace [["with resolved query" {:toucan2.pipeline/resolved-query {}}] ["with parsed args" {:toucan2.pipeline/query-type :toucan.query-type/select.instances.fns, :toucan2.pipeline/parsed-args {:kv-args {:key "ldap-sync-admin-group"}, :queryable {}}}] ["with model" {:toucan2.pipeline/model :setting}] ["with unparsed args" {:toucan2.pipeline/query-type :toucan.query-type/select.instances.fns, :toucan2.pipeline/unparsed-args (:setting :key "ldap-sync-admin-group")}] ["resolve connection" {:toucan2.connection/connectable org.h2.jdbc.JdbcConnection}] ["resolve connection" {:toucan2.connection/connectable nil}]]}
	at clojure.tools.logging$eval136$fn__139.invoke(NO_SOURCE_FILE:0)
	at clojure.tools.logging.impl$fn__1684$G__1666__1695.invoke(impl.clj:16)
	at clojure.tools.logging$log_STAR_.invokeStatic(logging.clj:63)
	at clojure.tools.logging$log_STAR_.invoke(logging.clj:42)
	at toucan2.honeysql2$apply_kv_arg_primary_method_default_clojure_lang_IPersistentMap_default.invokeStatic(honeysql2.clj:47)
	at toucan2.honeysql2$apply_kv_arg_primary_method_default_clojure_lang_IPersistentMap_default.invoke(honeysql2.clj:44)
	at clojure.lang.AFn.applyToHelper(AFn.java:171)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:675)
...
```

--- a/deps.edn
+++ b/deps.edn
@@ -74,7 +74,7 @@
   info.sunng/ring-jetty9-adapter            {:mvn/version "0.22.3"}             ; Drop-in replacement for official Ring Jetty adapter. Supports Jetty 11 webserver.
   instaparse/instaparse                     {:mvn/version "1.4.12"}             ; Make your own parser
   clj-commons/clj-yaml                      {:mvn/version "1.0.27"}             ; Clojure wrapper for YAML library SnakeYAML
-  io.github.camsaul/toucan2                 {:mvn/version "1.0.535"}
+  io.github.camsaul/toucan2                 {:mvn/version "1.0.537"}
   io.github.eerohele/pp                     {:git/tag "2023-11-25.47"           ; super fast pretty-printing library
                                              :git/sha "d3210d3"
                                              :git/url "https://github.com/eerohele/pp"}

